pkg/deployments/manager.go

```diff
@@ -69,33 +69,41 @@ func (r *Manager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
 		return ctrl.Result{}, fmt.Errorf("get: %w", err)
 	}
 
-	if ann := d.GetAnnotations(); ann != nil {
-		modelCSV, ok := ann[lingoDomain+"/models"]
-		if !ok {
-			return ctrl.Result{}, nil
-		}
-		models := strings.Split(modelCSV, ",")
-		if len(models) == 0 {
-			return ctrl.Result{}, nil
-		}
-		for _, model := range models {
-			r.setModelMapping(strings.TrimSpace(model), d.Name)
-		}
-	}
+	return ctrl.Result{}, r.addDeployment(ctx, d)
+}
 
+func (r *Manager) addDeployment(ctx context.Context, d appsv1.Deployment) error {
+	models := getModelsFromAnnotation(d.GetAnnotations())
```

Comment:
This method is not called twice on startup for each deployment. The code be optimized to exit early for know deployments